src/tool_operate.c

```diff
@@ -1575,6 +1581,12 @@ static CURLcode single_transfer(struct GlobalConfig *global,
           my_setopt(curl, CURLOPT_XFERINFOFUNCTION, tool_progress_cb);
           my_setopt(curl, CURLOPT_XFERINFODATA, &per->progressbar);
         }
+        else if(per->uploadfile && !strcmp(per->uploadfile, ".")) {
+          /* when reading from stdin in non-blocking mode, we use the progress
+             function to unpause a busy read */
+          my_setopt(curl, CURLOPT_XFERINFOFUNCTION, tool_readbusy_cb);
+          my_setopt(curl, CURLOPT_XFERINFODATA, per);
```

Comment:
I'm not happy with how this will
1. disable the regular default progress bar
2. disable the --progress-bar option if set
3. Not work for parallel transfers

... but I also don't have a very good answer on how to do this instead! :cry:

Reply:
I definitely agree.  I'm happy to explore other approaches, but I couldn't think of a better way. I wanted to put this out there to see what you guys thought.

Reply:
Regarding #2 above, It will actually honor the `--progress-bar` option.  I just pushed an update that adds the same busy read checking into the regular `CURLOPT_XFERINFOFUNCTION` handler also.

Reply:
Regarding 3 above.  I added the busy read checking to the parallel `CURLOPT_XFERINFOFUNCTION` handler also.

Reply:
Regarding 1 above.  I added a return code to the progress callback that will cause libcurl to continue executing the default progress function.